pkg/addons/seaweedfs/install.go

```diff
@@ -100,6 +100,16 @@ func createService(ctx context.Context, kcli client.Client, serviceCIDR string)
 
 	obj.ObjectMeta.Labels = ApplyLabels(obj.ObjectMeta.Labels, "s3")
 
+	var existingObj corev1.Service
+	if err := kcli.Get(ctx, client.ObjectKey{Name: obj.Name, Namespace: obj.Namespace}, &existingObj); err != nil && !k8serrors.IsNotFound(err) {
+		return errors.Wrap(err, "get s3 service")
+	} else if err == nil {
+		err := kcli.Delete(ctx, &existingObj)
```

Comment:
```suggestion
	} else if err == nil {
		if existingObj.Spec.ClusterIP == clusterIP {
			return nil
		fi
		err := kcli.Delete(ctx, &existingObj)
```

We shouldn't delete+recreate if it's already fine